core/DataAccess/ArchiveWriter.php

```diff
@@ -152,7 +164,12 @@ public function finalizeArchive()
         $numericTable = $this->getTableNumeric();
         $idArchive    = $this->getIdArchive();
 
-        $this->getModel()->updateArchiveStatus($numericTable, $idArchive, $this->doneFlag, self::DONE_OK);
+        $doneValue = $this->parameters->isPartialArchive() ? self::DONE_PARTIAL : self::DONE_OK;
+        $this->getModel()->updateArchiveStatus($numericTable, $idArchive, $this->doneFlag, $doneValue);
+
+        if ($this->parameters->isPartialArchive()) {
```

Comment:
@diosmosis shouldn't this be more like `if (!$this->parameters->isPartialArchive()) {`? If it's only a partial archive, can we really delete older archives? Or maybe I'm missing something?

Reply:
also do we need to check if `$this->earliestNow` is set?

Reply:
👍 will add a test covering both cases